schema/stacks/stack_schema_merge.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	"path/filepath"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	tfaddr "github.com/hashicorp/terraform-registry-address"
+	tfmod "github.com/hashicorp/terraform-schema/module"
+	tfschema "github.com/hashicorp/terraform-schema/schema"
+	"github.com/hashicorp/terraform-schema/stack"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type StackSchemaMerger struct {
+	coreSchema   *schema.BodySchema
+	stateReader  StateReader
+	moduleReader ModuleReader
```

Comment:
What do you think about combining both readers into a single `StateReader`? So that terraform-schema doesn't need to know the implementation details of the language server?